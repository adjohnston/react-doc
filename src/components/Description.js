import React from 'react'
import PropTypes from 'prop-types'
import classString from 'helpers/class-string'

const Description = ({children: description}) => ((!!description && (
  <section
    className={classString('__description')}>
    {description}
  </section>
)) || null)

Description.propTypes = {
  children: PropTypes.node
}

export default Description
