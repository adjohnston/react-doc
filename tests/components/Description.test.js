import React from 'react'
import renderer from 'react-test-renderer'
import Description from 'src/components/Description'

test('it should render correctly with children as a string', () => {
  const mockProps = {
    children: 'I am just a basic description'
  }

  const tree = renderer
    .create(<Description {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly with children as a component', () => {
  const mockProps = {
    children: (
      <p>
        'I am just a basic description component'
      </p>
    )
  }

  const tree = renderer
    .create(<Description {...mockProps} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render null with no children', () => {
  const tree = renderer
    .create(<Description />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
