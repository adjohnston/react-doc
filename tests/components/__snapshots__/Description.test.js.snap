// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly with children as a component 1`] = `
<section
  className="stncl__description"
>
  <p>
    'I am just a basic description component'
  </p>
</section>
`;

exports[`it should render correctly with children as a string 1`] = `
<section
  className="stncl__description"
>
  I am just a basic description
</section>
`;

exports[`it should render null with no children 1`] = `null`;
